addons/website_slides/static/src/js/public/components/slide_quiz_finish_dialog/slide_xp_progress_bar.js

```diff
@@ -0,0 +1,100 @@
+/** @odoo-module **/
+
+import { Component, onMounted, useState } from "@odoo/owl";
+import { browser } from "@web/core/browser/browser";
+
+export class SlideXPProgressBar extends Component {
+    static props = { rankProgress: Object };
```

Comment:
Actually, wouldn't it simplify a lot below to have this instead?

```suggestion
    static props = { 
        previous_rank: Object, 
        next_rank: Object,
        level_up: Boolean
    };
```